packages/components/src/components/expansion-panel/expansion-panel.jsx

```diff
@@ -0,0 +1,102 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import Icon from '../icon';
+
+class ExpansionPanel extends React.Component {
+    state = {
+        open_id: [],
+        isOpen: false,
+    };
+
+    onClick = () => {
+        // close if clicking the expansion panel that's open, otherwise open the new one
+        this.setState({ isOpen: !this.state.isOpen });
+    };
+
+    onArrayItemClick = id => {
+        this.setState(state => {
+            if (state.open_id.includes(id)) {
+                const open_id = state.open_id.filter(item => {
+                    return item !== id;
+                });
+                return {
+                    ...state,
+                    open_id,
+                };
+            }
+            return {
+                ...state,
+                open_id: [...state.open_id, id],
+            };
+        });
+    };
+
+    renderArrayContent = array => {
+        return (
+            <div>
+                {array.map((item, index) => {
+                    if (Array.isArray(item.value)) {
+                        return (
+                            <div key={index} className='dc-expansion-panel__content-array'>
+                                <span className='dc-expansion-panel__content-array-item-index'>{index + 1}: </span>(
+                                {item.value.length})
+                                {this.state.open_id.includes(item.id) ? (
+                                    <Icon
+                                        className='dc-expansion-panel__content-chevron-icon'
+                                        icon='IcChevronDown'
+                                        onClick={() => this.onArrayItemClick(item.id)}
+                                    />
+                                ) : (
+                                    <Icon
+                                        className='dc-expansion-panel__content-chevron-icon'
+                                        icon='IcChevronRight'
+                                        onClick={() => this.onArrayItemClick(item.id)}
+                                    />
```

Comment:
i think you can use css `transform: rotate(180deg)` to rotate the arrow here. We can apply animation using `transition` as well. What do you think?